code/game/mecha/equipment/tools/tools.dm

```diff
@@ -14,15 +13,27 @@
 
 /obj/item/mecha_parts/mecha_equipment/tool/hydraulic_clamp/attach(obj/mecha/M as obj)
 	..()
-	cargo_holder = M
+	if(istype(chassis, /obj/mecha/working/ripley))
+		var/obj/mecha/working/ripley/R = chassis
+		R.hydraulic_clamp = src
+	return
+
+/obj/item/mecha_parts/mecha_equipment/tool/hydraulic_clamp/detach()
+	..()
+	if(istype(chassis, /obj/mecha/working/ripley))
+		var/obj/mecha/working/ripley/R = chassis
+		R.hydraulic_clamp = null
 	return
 
 /obj/item/mecha_parts/mecha_equipment/tool/hydraulic_clamp/action(atom/target)
 	if(!action_checks(target)) return
-	if(!cargo_holder) return
-	if(istype(target, /obj/structure/stool)) return
+	if(!istype(chassis, /obj/mecha/working/ripley)) return
+	var/obj/mecha/working/ripley/R = chassis
+
+	if(istype(target, /obj/structure/stool))
+		return //no picking up rollerbeds
 	for(var/M in target.contents)

```

Comment:
Change this to if(target.search_contents_for(/mob/living))
Damn you, oldcoders.